### PR TITLE
documentation: fix typo in RegisterCodec godoc

### PR DESCRIPTION
**Pull request by RobinLG**
modify Compressors to Codecs

RELEASE NOTES: None

### CODE REVIEW

File: encoding/encoding.go [hunk 1]
```diff
@@ -108,7 +108,7 @@ var registeredCodecs = make(map[string]Codec)
 // more details.
 //
 // NOTE: this function must only be called during initialization time (i.e. in
-// an init() function), and is not thread-safe.  If multiple Compressors are
+// an init() function), and is not thread-safe.  If multiple Codecs are
```
Comment (menghanl) [inline encoding/encoding.go:111]: This doesn't seem like the right change. This is to register codecs, not compressors.

Comment (RobinLG) [reply encoding/encoding.go:111]: Line 72 is the same as line 111. I think line 72 is right, but the comment is for RegisterCodec(...) in line111.
And like you said "This is to register codecs, not compressors." so, the comment should be changed codecs?

Comment (RobinLG) [reply encoding/encoding.go:111]: > This doesn't seem like the right change. This is to register codecs, not compressors.
Line 72 is the same as line 111. I think line 72 is right, but the comment is for RegisterCodec(...) in line111.
And like you said "This is to register codecs, not compressors." so, the comment should be changed codecs?

Comment (RobinLG) [reply encoding/encoding.go:111]: > Line 72 is the same as line 111. I think line 72 is right, but the comment is for RegisterCodec(...) in line111.
And like you said "This is to register codecs, not compressors." so, the comment should be changed codecs?

Comment (menghanl) [reply encoding/encoding.go:111]: Line 72 is for function `RegisterCompressor`, and line 111 is for `RegisterCodec`

Comment (RobinLG) [reply encoding/encoding.go:111]: However, it seem like we can register compressors to `RegisterCodec` according to the unmodified comments.

Actually we can only register codecs to `RegisterCodec` according to the code.

Comment (menghanl) [reply encoding/encoding.go:111]: Ohhhh, sorry. I see. I read it wrong.
Yeah this look like the right fix. Thanks for the change!

**Review by menghanl — APPROVED** [review]

**Pull request merged**